### ISSUE TITLE
Suggestions for training on multiple environments simultaneously?

### ISSUE THREAD

**Issue by fedshyvana** [MarcoMeter/recurrent-ppo-truncated-bptt#7]
Hi, thanks for making this nice repo! So I am interested in testing whether recurrent policy gradients can be used for image classification (sounds strange but basically each image is extremely large like 100,000 by 100,000 so it's impossible to process the entire image at once - instead I think perhaps I can train an agent to selectively navigate parts of the image sequentially, pick-up on landmarks of interests, and then make a classification). 

So naturally, the goal here is not to solve a single environment (i.e. one image) since the model will just memorize the label. The goal is to train on e.g. 10,000 different environments (images), and test how well the agent + classifier will perform on held-out data. Do you have any suggestions for whether it makes sense to use your repo as a starting point, and modify it such that we either use a new environment in each mini-batch or sample from multiple environments in the same mini-batch? thanks!

**Comment by MarcoMeter** [comment]
Hi @fedshyvana 
you just have to define your environment as a gym. Just like how it is done by the provided environments.
All the logic you need can be done inside your environment. Like once an episode completed sample another image for the next episode.

**Comment by fedshyvana** [comment]
@MarcoMeter thanks! That makes perfect sense. On a related note, if I understand correctly, for long episodes, your implementation enables truncated backpropagation, where the number of steps to backpropagate is controlled by **sequence_length**. Effectively, in the implementation if **sequence_length** = K, an episode of length L will be split into L - K + 1 sequences:
e.g. L = 4, K = 2, and episode is [0,1,2,3]:
episode will be split into 3 sequences: [0, 1], [1, 2], [2, 3], the so the input goes from 1 x 4 x features ---> 3 x 2 x features before it's fed to the recurrent policy network. Is this correct? It's a little confusing to me that we're using truncated backpropagation to save computation/memory requirement but we're actually making the input size in each mini-batch larger...  does the reduction in backward computation outweighs the increased memory/compute required for the forward pass in this case? 

Thanks!

**Comment by MarcoMeter** [comment]
Given your example, the episode of [0,1,2,3] would be split into two sequences: [0,1] and [2,3].
If you don't want to use truncated bptt, you can set the `sequence_length` to `-1`.
The longer the sequence length, the greater the memory cost of the gradients will be.
